File: src/main/java/com/bajdcc/util/lexer/regex/RegexStringIteratorData.java
package com.bajdcc.util.lexer.regex;

import com.bajdcc.util.lexer.token.MetaType;

/**
 * 分析时使用的数据
 */
public class RegexStringIteratorData implements Cloneable {
	/**
	 * 当前处理的位置
	 */
	public int iIndex = 0;

	/**
	 * 字符
	 */
	public char chCurrent = 0;

	/**
	 * 字符类型
	 */
	public MetaType kMeta = MetaType.END;

	public RegexStringIteratorData() {

	}

	public RegexStringIteratorData(int index, char current, MetaType meta) {
		iIndex = index;
		chCurrent = current;
		kMeta = meta;
	}

	@Override
	public Object clone() throws CloneNotSupportedException {
		return super.clone();
	}
}

File: src/main/java/com/bajdcc/util/lexer/regex/RegexStringIteratorData.kt
package com.bajdcc.util.lexer.regex

import com.bajdcc.util.lexer.token.MetaType

/**
 * 分析时使用的数据
 * [index] 当前处理的位置
 * [current] 字符
 * [zero] 允许空字符
 * [meta] 字符类型
 */
data class RegexStringIteratorData(var index: Int = 0,
                                   var current: Char = 0.toChar(),
                                   var zero: Boolean = false,
                                   var meta: MetaType = MetaType.END) : Cloneable {

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): RegexStringIteratorData {
        return super.clone() as RegexStringIteratorData
    }
}